lib/assert.js

```diff
@@ -15,13 +15,16 @@ const noop = () => {};
 Object.defineProperty(x, 'AssertionError', {value: assert.AssertionError});
 
 function create(val, expected, operator, msg, fn) {
-	return {
-		actual: val,
-		expected,
-		message: msg,
-		operator,
-		stackStartFunction: fn
-	};
+	if (msg === undefined || typeof msg === 'string') {
+		return {
+			actual: val,
+			expected,
+			message: msg,
+			operator,
+			stackStartFunction: fn
+		};
+	}
+	throw new TypeError('assertion message must be a string');
```

Comment:
I can check if msg exists and not string then throw an error in within the if-statement:

`
if(msg && typeof msg !== 'string'){
  throw error 
}
` 

But that logic is flawed because 0 will slip through uncaught. Also, inverting the logic as suggested seem flawed as well since an OR condition is short-circuited once first test is truthy.

I'll look more closely.

Cheers!